src/Adapter/Shop/MaintenanceConfiguration.php

```diff
@@ -44,10 +44,12 @@ class MaintenanceConfiguration extends AbstractMultistoreConfiguration
      */
     public function getConfiguration()
     {
+        $shopConstraint = $this->getShopConstraint();
+
         return [
             'enable_shop' => $this->configuration->getBoolean('PS_SHOP_ENABLE'),
-            'maintenance_ip' => $this->configuration->get('PS_MAINTENANCE_IP'),
-            'maintenance_text' => $this->configuration->get('PS_MAINTENANCE_TEXT'),
+            'maintenance_ip' => $this->configuration->get('PS_MAINTENANCE_IP', null, $shopConstraint),
```

Comment:
The problem is that now `getBoolean` doesn't use the shop constraint This makes higher risks of having differences. So either we update the `getBoolean` function but I'm not sure you can because of the inherited `ParameterBag`

Or you handle the casting alone and only use the `get` method. So maybe we should deprecate the `getBoolean/getInt/...` methods, or add them additional parameters for consistency.

Reply:
I think @eternoendless would tell us to stop relying on broken adapters and use legacy class directly.